tools/cldr-apps/src/main/java/org/unicode/cldr/web/CookieSession.java

```diff
@@ -34,926 +33,1012 @@
+        );
+        return false;
+      }
     }
-
-    public static final String cheapDecodeString(String s) {
-        return new String(cheapDecode(s), StandardCharsets.UTF_8);
-    }
-
-    public static final byte[] cheapDecode(String s) {
-        return Base64.getUrlDecoder().decode(s);
+  }
+
+  // parameters
+
+  /**
+   * last time reaped. Starts at 0, so reap immediately
+   */
+  static long lastReapMillisSinceEpoch = 0;
+
+  /**
+   * Number of guests
+   */
+  private static int nGuests = 0;
+
+  /**
+   * Number of users
+   */
+  private static int nUsers = 0;
+
+  public static int getGuestCount() {
+    getUserCount();
+    return nGuests;
+  }
+
+  /**
+   * Perform a reap if need be, and count.
+   * @return user count
+   */
+  public static int getUserCount() {
+    synchronized (uHash) {
+      return uHash.size();
     }
-
-    /**
-     * Parameters for when to disconnect users.
-     * These defaults can be changed in cldr.properties.
-     * Formerly CLDR_USER_TIMEOUT only related to what's now called KICK_IF_ABSENT,
-     * while CLDR_USER_INACTIVITY only related to what's now called KICK_IF_INACTIVE.
-     * No more distinction _INACTIVITY versus _TIMEOUT, use _TIMEOUT_SECS for all.
-     * Reference: https://unicode-org.atlassian.net/browse/CLDR-11799
-     *
-     * @author srl
-     */
-    private enum Params {
-        /**
-         * max logged in user count allowed before stricter requirements apply to stay connected
-         */
-        CLDR_MAX_USERS(30),
-
-        /**
-         * max guests allowed before guests start getting shut out (should be < CLDR_MAX_USERS)
-         */
-        CLDR_MAX_GUESTS(0), // zero means tooManyGuests == tooManyUsers
-
-        /**
-         * how many seconds guest can be absent/inactive before getting kicked
-         */
-        CLDR_GUEST_TIMEOUT_SECS(5 * 60), // 5 minutes
-
-        /**
-         * how many seconds logged-in user can be absent/inactive before getting kicked
-         */
-        CLDR_USER_TIMEOUT_SECS(30 * 60); // 30 minutes
-
-        private int defVal;
-        private Integer value;
-
-        /**
-         * Construct a new parameter
-         * @param name
-         * @param defVal
-         */
-        Params(int defVal) {
-            this.defVal = defVal;
+  }
+
+  private static int lastCount = -1;
+
+  public static int checkForExpiredSessions() {
+    synchronized (gHash) {
+      int allCount = gHash.size(); // count of ALL users
+      long nowMillisSinceEpoch = System.currentTimeMillis();
+      long elapsedMillis = (nowMillisSinceEpoch - lastReapMillisSinceEpoch);
+
+      final boolean tooManyUsers = tooManyUsers();
+      final boolean tooManyGuests = tooManyGuests();
+
+      final long CHECK_SECS = 5; // check every 5 seconds or if count grows
+      if (
+        elapsedMillis < (1000 * CHECK_SECS) &&
+        allCount <= lastCount &&
+        !tooManyGuests
+      ) {
```

Comment:
this, however, is definitely an improvement (though the expression is more complex than it ought to be)